odl/solvers/functional/default_functionals.py

```diff
@@ -59,18 +60,23 @@ class LpNorm(Functional):
         \| x \|_p = \\left(\\int_\Omega |x(t)|^p dt. \\right)^{1/p}
     """
 
-    def __init__(self, space, exponent):
+    def __init__(self, domain, exponent, range=None):
         """Initialize a new instance.
 
         Parameters
         ----------
-        space : `DiscreteLp` or `TensorSpace`
+        domain : `DiscreteLp` or `TensorSpace`
             Domain of the functional.
         exponent : float
             Exponent for the norm (``p``).
+        range : `Field`, optional
+            Range of the functional. Default: `RealNumbers`.
         """
+        if range is None:
+            range = RealNumbers()
```

Comment:
How about 
```python
def __init__(self, domain, exponent, range=RealNumbers()):
    ...
```
? I feel that the current variant is unnecessarily indirect. `None` is usually a proxy for `some_value_that_depends_on_other_stuff_or_must_be_computed`

Reply:
This only works if we want `RealNumbers` to be forever immutable. Are we sure about that? (Id be fine with it)

Reply:
Yeah, we can make that clearer by adding `__slots__` to `RealNumbers` and `ComplexNumbers`.